winbooks-api-extra/src/main/java/be/valuya/winbooks/api/extra/WinbooksExtraService.java

```diff
@@ -147,6 +147,44 @@ public int getAccountNumberLengthFromParamsTable(WinbooksFileConfiguration winbo
                 .orElse(ACCOUNT_NUMBER_DEFAULT_LENGTH);
     }
 
+    /**
+     * Creates directory hierarchy, honoring config case-insensitive settings, and preventing ftp errors on exsting
+     * directories.
+     *
+     * @param fileConfiguration
+     * @param path
+     */
+    public void createDirectories(WinbooksFileConfiguration fileConfiguration, Path path) {
+        // To handle ftp errors, create directories at each level after checking it does not exist yet
+        int pathNameCount = path.getNameCount();
+        boolean resolveCaseInsensitiveSiblings = fileConfiguration.isResolveCaseInsensitiveSiblings();
+        Path curPath = path.getFileSystem().getRootDirectories().iterator().next();
+
+        for (int nameIndex = 0; nameIndex < pathNameCount; nameIndex++) {
+            Path nextName = path.getName(nameIndex);
+            Optional<Path> resolvedPathOptional = WinbooksPathUtils.resolvePath(curPath, nextName.toString(), resolveCaseInsensitiveSiblings);
+
+            if (resolvedPathOptional.isPresent()) {
+                Path resolvedPath = resolvedPathOptional.get();
```

Comment:
Please avoid get. `if (isPresent())` `get()` / `else` can be replaced by `Optional::ifPresentOrElse`.